### PR TITLE
Added snmpv3

### PR DESCRIPTION
**Pull request by yaroslavNqualisystems**
This change is [Reviewable](https://reviewable.io/reviews/qualisystems/cloudshell-networking-juniper/53)

### CODE REVIEW

File: cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py [hunk 1]
```diff
@@ -19,13 +21,35 @@ def __init__(self, cli_handler, logger):
...
+            raise Exception(self.__class__.__name__, "SNMP Community has to be defined")
+        snmp_actions = EnableDisableSnmpActions(cli_service, self._logger)
+        commit_rollback = CommitRollbackActions(cli_service, self._logger)
+        try:
+            self._logger.debug('Disable SNMP')
+            snmp_actions.disable_snmp(snmp_community)
+            commit_rollback.commit()
+        except CommandExecutionException as exception:
+            commit_rollback.rollback()
+            self._logger.error(exception)
```
Comment (anthony-poddubny) [inline cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py]:
```suggestion
             self._logger.exception("Failed to disable SNMP")
```

File: cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py [hunk 2]
```diff
@@ -19,13 +21,35 @@ def __init__(self, cli_handler, logger):
...
+        snmp_v3_actions = EnableDisableSnmpV3Actions(cli_service, self._logger)
+        commit_rollback = CommitRollbackActions(cli_service, self._logger)
+        snmp_user = snmp_parameters.snmp_user
+        try:
+            self._logger.debug('Disable SNMPv3')
+            snmp_v3_actions.disable_snmp_v3(snmp_user)
+            commit_rollback.commit()
+        except CommandExecutionException as exception:
+            commit_rollback.rollback()
+            self._logger.error(exception)
```
Comment (anthony-poddubny) [inline cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py]:
```suggestion
             self._logger.exception("Failed to disable SNMPv3")
```

File: cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py [hunk 3]
```diff
@@ -19,31 +20,53 @@ def __init__(self, cli_handler, logger):
...
-                    else:
-                        output = snmp_actions.enable_snmp(snmp_community, write=False)
-                    output += commit_rollback.commit()
-                    return output
-                except CommandExecutionException as exception:
-                    commit_rollback.rollback()
-                    self._logger.error(exception)
-                    raise exception
+        if not snmp_community:
+            raise Exception(self.__class__.__name__, "SNMP Community has to be defined")
```
Comment (anthony-poddubny) [inline cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py]:
```suggestion
             raise Exception("SNMP Community has to be defined")
```

File: cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py [hunk 4]
```diff
@@ -19,31 +20,53 @@ def __init__(self, cli_handler, logger):
...
+        if not snmp_actions.configured(snmp_community):
+            self._logger.debug('Configuring SNMP with community {}'.format(snmp_community))
+            try:
+                output = snmp_actions.enable_snmp(snmp_community,
+                                                  write=isinstance(snmp_parameters, SNMPV2WriteParameters))
+                output += commit_rollback.commit()
+                return output
+            except CommandExecutionException as exception:
+                commit_rollback.rollback()
+                self._logger.error(exception)
```
Comment (anthony-poddubny) [inline cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py]:
```suggestion
                 self._logger.exception("Failed to configure SNMP")
```

File: cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py [hunk 5]
```diff
@@ -19,31 +20,53 @@ def __init__(self, cli_handler, logger):
...
+        snmp_priv_proto = snmp_parameters.private_key_protocol
+        self._logger.debug('Enable SNMPv3')
+        try:
+            output = snmp_v3_actions.enable_snmp_v3(snmp_user, snmp_password, snmp_priv_key, snmp_auth_proto,
+                                                    snmp_priv_proto)
+            commit_rollback.commit()
+            return output
+        except CommandExecutionException as exception:
+            commit_rollback.rollback()
+            self._logger.error(exception)
```
Comment (anthony-poddubny) [inline cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py]:
```suggestion
             self._logger.exception("failed to enable SNMPv3")
```

Comment (yaroslavNqualisystems) [reply cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py]: done

File: cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py (hunk 4 above)
Comment (yaroslavNqualisystems) [reply cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py]: done

File: cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py (hunk 3 above)
Comment (yaroslavNqualisystems) [reply cloudshell/networking/juniper/flows/juniper_enable_snmp_flow.py]: done

File: cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py (hunk 2 above)
Comment (yaroslavNqualisystems) [reply cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py]: done

File: cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py (hunk 1 above)
Comment (yaroslavNqualisystems) [reply cloudshell/networking/juniper/flows/juniper_disable_snmp_flow.py]: done

**Review by anthony-poddubny — APPROVED** [review]

**Pull request merged**